Trustroots/trustroots
# Add RTL language support

#### Proposed Changes

* Adds Right-to-Left language support, relevant for locales like Arabic, Hebrew, Pashto, Persian, Sindhi, Syriac, Dhivehi, Urdu, Yiddish, etc.

See https://www.w3.org/International/questions/qa-html-dir

#### Testing Instructions

Run `npm run build` (as building itself doesn't currently work on dev build, needs a bit more work but I think it can be separate PR)

After first run, everything should work on both `npm start` and `npm run start:prod`

* Open language switcher and switch to Hebrew
    
    <img width="1002" alt="image" src="https://user-images.githubusercontent.com/87168/117666873-f9e5a480-b1ac-11eb-9695-7736c90451da.png">
* Page turns flipped right-to-left:
    
    ![image](https://user-images.githubusercontent.com/87168/119261667-1b994f80-bbe1-11eb-99dd-8b3c925fd193.png)


* Notice different CSS tag getting appended at the end of `<head>`; if you switch between multiple languages, the CSS ever loads just once and isn't removed in between.
* Notice `lang` and `dir` get updated in HTML tag in DOM. `dir="ltr"` puts loaded CSS in affect. 
* Refresh the page, RTL CSS should remain. This is because language code is stored in your browser as a cookie and in user settings when logged in. On initialization, the CSS gets loaded again.
    * Ideally this would be from server-side but client-side is fine for now methinks
* There is more work to do how CSS rendering looks like but that's good for follow-up. 
  
Part of https://github.com/Trustroots/trustroots/issues/1485
